warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexCachingIteratorJexl.java

```diff
@@ -865,16 +881,34 @@ public boolean add(long val) {
 
     private void fillSortedSets() throws IOException {
         String sourceRow = this.fiRow.toString();
-        setupRowBasedHdfsBackedSet(sourceRow);
-
-        // if keys is not null, then we already had a completed set which was loaded in setupRowBasedHdfsBackedSet
-        if (keys != null) {
-            moveToNextRow();
-            return;
+        // if we are running fillSortedSets as port of a re-seek after yield, then the fillSet threads would not have
+        // completed when a WaitWindowOverrunException was thrown and therefore the set would not have been marked as
+        // complete when setupRowBasedHdfsBackedSet was called at the top of this method. We will try to copy the
+        // RowBasedHdfsBackedSet from the previously used Ivarator since there may be unpersisted results or
+        // a long-running Ivarator may still be filling the RowBasedHdfsBackedSet
+        // If this re-seek is from an Ivarator yielding, the startKey will have a yield marker in either the
+        // colFam (for a shard range) or colQual (for a document range)
+        boolean usePreviousSortedSet = WaitWindowObserver.hasMarker(lastRangeSeeked.getStartKey());
```

Comment:
I think we are missing the case where the ivarator filled the set, and the tablet migrates.  We need to think through those scenarios.